homeworks/jonasnikolett/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,136 @@
+"""Feladat 1: 
+Hozz létre egy to_do.py nevű file-t, és kódold le a következő feladat megoldását:
+Készíts egy Feladatkezelő alkalmazást! 
+Hozz létre egy .txt file-t és hagyd üresen.
+Definiálj 3 függvényt a következőkre: feladatok olvasása, egy feladat hozzáadása, egy feladat törlése.
+Legyen egy display_menu() function-öd is, ami kiprinteli a lehetséges opciókat:
+Add Task
+View Tasks
+Remove Task
+Exit
+Folyamatosan kérj be inputot a felhasználótól hogy ezek közül a menüpontok közül mit szeretne csinálni,
+és hívd meg a válaszhoz megfelelő függvényt.
+A felhasználó inputja 1,2,3 vagy 4 kell, hogy legyen, ellenőrizd! Ha az 1-es vagy 3-as opciót választja,
+mindkét esetben paramétert kell átadnod a megfelelő függvénynek. “Exit”-re lépjen ki a program.
+Használj hibakezelést a file-ba való íráskor és olvasáskor, illetve használd a logging module-t.
+Egyszerre logolj a konzolra és egy .log file-ba. A .txt file legyen része a pull request-nek,
+de a log file-ok ne! Használd a .gitignore-t!
+
+Extra: Alakítsd át a .txt file-os megoldást úgy, hogy egy json file-t használsz a feladatok trackelésére!
+"""
+#egyéb szükséges dolgok
+import os
+import logging
+
+#file elérési helye
+
+base_dir = os.path.dirname(os.path.abspath(__file__))
+data_file = os.path.join(base_dir, 'jonasnikolett.txt')
+log_file = os.path.join(base_dir, 'to_do_log.log')
+
+#loggolás meghatározása
+stream_handler = logging.StreamHandler()
+stream_handler.setLevel(logging.WARNING)
+
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+stream_handler.setFormatter(formatter)
+
+file_handler = logging.FileHandler(log_file)
+file_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+file_handler.setFormatter(formatter)
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+logger.addHandler(stream_handler)
+logger.addHandler(file_handler)
+
+
+#file megnyitása és hozzáírás meghatározása
+def read_tasks(filename):
+    try:
+        with open(filename, 'r'):
+            tasks = filename.readlines()
```

Comment:
a filename itt is egy string paraméter, erre nem tudjuk megívni a readlines()-t. Helyesen így néz ki:

```
def read_tasks(filename):
    try:
        with open(filename, 'r') as f:
            tasks = f.readlines()
            logging.info("Registered task / Feladat beírásra került")
            return tasks
    except FileNotFoundError:
        logging.error("File not found! / A fájl nem található.")
        return []
```

Reply:
Megcsináltam a javítást, ahogy írtad. De nem értem mert nekem mikor lefuttattam az eredetit nem írt ki hibát. Ez mitől függ hogy kiírja-e hogy hibás vagy sem, de a hibás akkor miért futtatta le a gépem?